omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -114,6 +114,49 @@ public void shouldEditAPatientIdentifier() throws Exception {
 	}
 	
 	@Test
+	public void shouldEditAPatientIdentifierWithPreferredAttibuite() throws Exception {
+		PatientIdentifier patientIdentifier = service.getPatientIdentifierByUuid(getUuid());
+		Boolean originalPreferredStatus = patientIdentifier.getPreferred();
+		Boolean postPreferredStatus = !originalPreferredStatus;
+		SimpleObject postPatientIdentifier = new SimpleObject();
+		postPatientIdentifier.add("preferred", postPreferredStatus);
+		
+		String json = new ObjectMapper().writeValueAsString(postPatientIdentifier);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertNotEquals(originalPreferredStatus, patientIdentifier.getPreferred());
```

Comment:
After posting, how do you confirm that the preferred was persisted?

Reply:
line 156 assertEquals(preferredIdentifer.isPreferred(), true);

Reply:
assertNotEquals(originalPreferredStatus, patientIdentifier.getPreferred()); Assert that the new value is not the same with old one.

Reply:
I toggled the value on line 120 newPreferredStatus = !originalPreferredStatus;

Reply:
So assuming patientIdentifier.getPreferred() is TRUE, then posting with toggled value will make it FALSE. then I will assert that the new value which is now FALSE is no longer the original value(TRUE)